tc/examples/group_normalization.cc

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "tc/aten/aten.h"
+#include "tc/aten/aten_autotuner.h"
+#include "tc/aten/aten_compiler.h"
+#include "tc/autotuner/genetic_search.h"
+#include "tc/core/cpu/cpu_mapping_options.h"
+#include "tc/core/cpu/cpu_tc_executor.h"
+#include "tc/core/cuda/cuda_mapping_options.h"
+#include "tc/core/cuda/cuda_tc_executor.h"
+#include "tc/core/flags.h"
+
+DEFINE_string(proto_path, "", "Filename to load and store proto cache ");
+DEFINE_bool(
+    use_best_options,
+    false,
+    "Start from hardcoded best options; if false start from naive options ");
+
+auto previouslyTunedBestOptions =
+    tc::CudaMappingOptions::makeNaiveMappingOptions()
+        .outerScheduleFusionStrategy(tc::FusionStrategy::Max)
+        .outerScheduleAllowSkewing(false)
+        .outerSchedulePositiveOrthant(true)
+        .intraTileScheduleFusionStrategy(
+            tc::FusionStrategy::Preserve3Coincident)
+        .intraTileScheduleAllowSkewing(false)
+        .intraTileSchedulePositiveOrthant(true)
+        .tile(6, 1, 24)
+        .unroll(16)
+        .tileImperfectlyNested(false)
+        .matchLibraryCalls(false)
+        .mapToThreads(48, 6)
+        .mapToBlocks(256, 32)
+        .useSharedMemory(true)
+        .usePrivateMemory(true)
+        .unrollCopyShared(false);
+
+template <typename Backend>
+at::Tensor makeATenTensor(at::ArrayRef<long int> sizes);
+
+template <>
```

Comment:
This code keeps proliferating, I suggested to move it somewhere in the commit that introduced the first version.  I understand we want self-contained examples, but having a local `.h` is self-contained enough for C++ code IMO.